presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java

```diff
@@ -514,11 +519,21 @@ else if (type instanceof DecimalType) {
         }
     }
 
+    private Object getPrimitiveObjectValue(int column, Object fieldData)
+    {
+        try {
+            return ((PrimitiveObjectInspector) fieldInspectors[column]).getPrimitiveJavaObject(fieldData);
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(HIVE_BAD_DATA, String.format("Error parsing field value '%s' for field %s: %s", fieldData, column, e.getMessage()), e);
+        }
+    }
+
     private void validateType(int fieldId, Class<?> type)
     {
         if (!types[fieldId].getJavaType().equals(type)) {
             // we don't use Preconditions.checkArgument because it requires boxing fieldId, which affects inner loop performance
```

Comment:
Let's remove this comment now, since we're throwing `PrestoException` and thus `checkArgument` would not apply.